docs/runtime/types/module-struct.md

```diff
@@ -0,0 +1,241 @@
+pub struct Module<T: Trait>(::std::marker::PhantomData<(T)>);
+```
+
+> *Notes:*
+>
+> * The attribute `#[cfg(feature = "std")]` tells the Rust compiler to only compile our struct when using along with the Rust standard library.
+> * [`structural_match`](https://github.com/rust-lang/rfcs/pull/1445) is used to indicates that instances of the `Module` struct can be used within patterns.
+> * `rustc_copy_clone_marker` indicates the `Module` struct is able to be either copied or cloned.
+
+## Functions and Implementations
+
+### Dispatchable Functions
+
+Dispatchable functions are written by runtime developers and maintain the logic around how to manipulate the blockchain state. Here we only have one function named `do_something` in template runtime module, the expanded code is nearly same with original code except the added `origin` type signature.
+
+```rust
+impl<T: Trait> Module<T> {
+    pub fn do_something(origin: T::Origin, something: u32) -> Result {
+        let who = ensure_signed(origin)?;
+        <Something<T>>::put(something);
+        Self::deposit_event(RawEvent::SomethingStored(something, who));
+        Ok(())
+    }
+}
+```
+
+`Module` struct implements the `Callable` trait in `srml-support` by specifying the associate type to be the module's `Call` enum. Above dispatchable functions will be exposed through this enum, go to its own page for more details about [`Call` enum]((runtime/types/call-enum.md)).
+
+```rust
+impl<T: Trait> ::srml_support::dispatch::Callable for Module<T> {
+    type Call = Call<T>;
+}
+```
+
+There is also a `dispatch` function that redirect the request to the module's `Call` enum:
+
+```rust
+impl<T: Trait> Module<T> {
+    #[doc(hidden)]
+    pub fn dispatch<D: ::srml_support::dispatch::Dispatchable<Trait=T>>(
+        d: D,
+        origin: D::Origin,
+    ) -> ::srml_support::dispatch::Result {
+        d.dispatch(origin)
+    }
+}
+```
+
+### Implementations for Storage
+
+The input codes for the `decl_storage!` procedure macro in template module are:
+
+```rust
+/// This module's storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as TemplateModule {
+        // Just a dummy storage item. 
+        // Here we are declaring a StorageValue, `Something` as a Option<u32>
+        // `get(something)` is the default getter which returns either the stored `u32` or `None` if nothing stored
+        Something get(something): Option<u32>;
+    }
+}
+```
+
+After expanded, we'll see a `Store` implementation for `Module` struct. The `Store` trait here is an abstract layer that places all the associate types for module's available storage items. `Module` struct implements this trait by specifying the concrete types like `Something<T>` struct.
+
+```rust
+trait Store {
+    type Something;
+}
+
+struct Something<T: Trait>(
+    self::sr_api_hidden_includes_decl_storage::hidden_include::storage::generator::PhantomData<
+        (T),
+    >,
+);
+
+impl<T: Trait> Store for Module<T> {
```

Comment:
You should simply reference that the Module type has multple implementations, like Store, and if they want to learn more about how that implementation is generated, they can look at the Storage type.

This applies to other impl too

Reply:
I keep the sub-sections, since it makes the structure a bit clear.